uk-multi-ab-testing-service/src/main/java/com/marshmallow/uk_multi_ab_testing/persistence/PersistentABTestParticipant.java

```diff
@@ -0,0 +1,43 @@
+package com.marshmallow.uk_multi_ab_testing.persistence;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Value;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbImmutable;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey;
+
+import java.util.UUID;
+
+@Value
+@Builder(toBuilder = true)
+@DynamoDbImmutable(
+    builder = PersistentABTestParticipant.PersistentABTestParticipantBuilder.class)
+public class PersistentABTestParticipant {
+
+    public static final String TABLE_NAME = "ab_test_participants";
+
+    /**
+     * ID of the resource - e.g. quote ID, policy ID, customer ID etc.
+     */
+    @NonNull
+    @Getter(onMethod_ = @DynamoDbPartitionKey)
+    UUID resourceId;
```

Comment:
Don't need to assume the format of the IDs. Customers use NanoIDs instead of UUIDs.
```suggestion
    String resourceId;
```